File: 121.best-time-to-buy-and-sell-stock.go
// @leet start
func maxProfit(prices []int) int {
  min := prices[0]
  max := 0
  for i := 1; i < len(prices); i++ {
    if prices[i] < min {
      min = prices[i]
    }
    if prices[i] - min > max {
      max = prices[i] - min
    }
  }
  return max
}
// @leet end

File: 121.best-time-to-buy-and-sell-stock.ts
// @leet start
function maxProfit(prices: number[]): number {
  if (prices.length < 2) return 0;
  let min = prices[0];
  let max = 0;
  for (let i = 1; i < prices.length; i++) {
    if (prices[i] < min) {
      min = prices[i];
    }
    if (prices[i] - min > max) {
      max = prices[i] - min;
    }
  }
  return max;
};
// @leet end
